### PR TITLE
jax.scipy.fft: error for unsupported norm argument

### LINKED ISSUE #22466: Wrong normalization in FFT methods
### Description

Summary: The discrete cosine transformation libraries normalize the transforms incorrectly.

I am aware that JAX's documentation states that only `norm="ortho"` is supported, however, this still requires the user to explicitly pass `norm=ortho`, which the documentation does not state. To be clear, jax gives incorrect results unless `norm="ortho"` is passed explicitly. Jax also doesn't seem to implement backward dct and forward idct correctly. If this is not supported yet, then JAX should fail early. The effect of these bugs is that interpolation with fft's without specifying additional flags gives incorrect results.

Minimal reproductions.

Bug 1.
```python
import numpy as np
from scipy.fftpack import dct as sdct
from scipy.fftpack import idct as sidct
from jax.scipy.fft import dct as jdct
from jax.scipy.fft import idct as jidct
import matplotlib.pyplot as plt

m = np.linspace(0, 2 * np.pi, 100)
n = np.linspace(0, 2 * np.pi, 5 * m.size)
f = m
normalize = np.sqrt(n.size / m.size)
sf = sidct(sdct(f, norm="ortho"), n=n.size, norm="ortho") * normalize
jf = jidct(jdct(f, norm="ortho"), n=n.size, norm="ortho") * normalize
# Doesn't implement ortho correctly unless specifed. Documentation says only ortho is supported. Bug 1.
jf_bug = jidct(jdct(f), n=n.size) * normalize

plt.scatter(m, f, label="truth", color="black")
plt.plot(n, sf, linestyle="-", linewidth=4, label="ortho scipy")
plt.plot(n, jf, linestyle="--", linewidth=2.5, label="ortho jax")
plt.plot(n, jf_bug, linestyle="--", linewidth=2.5, label="ortho jax bug")
plt.legend()
```
out

Ok, so now the user wonders whether the documentation is out of date, and tries to find out what is the normalization that JAX implements by default. They may conclude that default is `backward` on `dct` and `forward` on `idct`. Scipy's default is `backward` for both. This departure from `scipy` seems confusing given that this methods are in `jax.scipy.fft`.

More confusing, is that when we specify explicitly to do `backward` on `dct` and `forward` on `idct` for `jax`, the results change, so it seems JAX does not implement the options that is does by default correctly when they are specified explicitly.

Bug 2.
```python
import numpy as np
from scipy.fftpack import dct as sdct
from scipy.fftpack import idct as sidct
from jax.scipy.fft import dct as jdct
from jax.scipy.fft import idct as jidct
import matplotlib.pyplot as plt

m = np.linspace(0, 2 * np.pi, 100)
n = np.linspace(0, 2 * np.pi, 5 * m.size)
f = m
normalize = n.size / m.size
sf = sidct(sdct(f, norm="backward"), n=n.size, norm="forward") * normalize
jf_bug = jidct(jdct(f), n=n.size) * normalize  # matches sf, so must be doing backward then forward
# But this does not match the above 2.
jf_bug_2 = jidct(jdct(f, norm="backward"), n=n.size, norm="forward") * normalize

plt.scatter(m, f, label="truth", color="black")
plt.plot(n, sf, linestyle='-', linewidth=4, label="scipy backward forward")
# This plot matches the truth, but only because we multiplied by "normalize",
# if it was doing ortho then it should only match if we multiply by sqrt(normalize),
# hence the bug
plt.plot(n, jf_bug, linestyle='--', linewidth=2.5, label="ortho jax bug")
# If jax doesn't support backward and forward, then this should fail early with an error.
plt.plot(n, jf_bug_2, linestyle='--', linewidth=2.5, label="backward forward jax bug 2")
plt.legend()
```
out2.

Please let me know if I am missing something.

### System info (python version, jaxlib version, accelerator, etc.)

```
An NVIDIA GPU may be present on this machine, but a CUDA-enabled jaxlib is not installed. Falling back to cpu.
jax:    0.4.28
jaxlib: 0.4.28
numpy:  1.26.4
python: 3.10.14 | packaged by conda-forge | (main, Mar 20 2024, 12:45:18) [GCC 12.3.0]
jax.devices (1 total, 1 local): [CpuDevice(id=0)]
process_count: 1
platform: uname_result(system='Linux', node='fedora', release='6.9.8-200.fc40.x86_64', version='#1 SMP PREEMPT_DYNAMIC Fri Jul  5 16:20:11 UTC 2024', machine='x86_64')
```

### PR DESCRIPTION
**Pull request by jakevdp**
Fixes #22466

### CODE REVIEW

**Review by dfm — APPROVED** [review]
Looks good!
